waku/v2/peermanager/topic_event_handler.go

```diff
@@ -118,6 +120,37 @@ func (pm *PeerManager) handlerPeerTopicEvent(peerEvt relay.EvtPeerTopic) {
 	wps := pm.host.Peerstore().(*wps.WakuPeerstoreImpl)
 	peerID := peerEvt.PeerID
 	if peerEvt.State == relay.PEER_JOINED {
+		// Metadata verification is done only if a clusterID is specified
+		if pm.metadata != nil && pm.metadata.ClusterID() != 0 {
```

Comment:
Don't think this is the right place for this check.
Because for lightNodes, this topicEvent won't be applicable. This event is generated only for relay nodes where the node advertises that it is interested in pubsubTopic as part of gossipsub Hello packet.

We can refer to how go-libp2p-pubsub is doing something similar for helloPacket as soon as peer is connected. https://github.com/libp2p/go-libp2p-pubsub/blob/d13e24ddc9f288557732c9c02b143c09b86c6778/pubsub.go#L677-L708

They seem to be using connectionNotifier to communicate Hello packet by opening a stream. We could do the same and then wait for response and decide whether to continue or not?

Reply:
Good point. I'll move this to a connection notifier.

Reply:
Done! code looks nicer now as the peer manager no longer has anything to do with the metadata protocol